__tests__/app.test.js

```diff
@@ -60,13 +69,59 @@ describe("GET /api/articles/:article_id returns correct article", () => {
       msg: "An article with provided article ID does not exist",
     });
   });
-  test("returns 404 error if article number is non-number", async () => {
-    const results = await request(app).get("/api/articles/cheese").expect(404);
+  test("returns 400 error if article number is non-number", async () => {
+    const results = await request(app).get("/api/articles/cheese").expect(400);
+    expect(results.body.msg).toBe("Bad Request");
+  });
+  test("returns 404 error if endpoint incorrectly inputted", async () => {
+    const results = await request(app).get("/api/article/2").expect(404);
     expect(results.body).toEqual({
-      msg: "An article with provided article ID does not exist",
+      msg: "Page not Found",
     });
   });
+});
+describe("PATCH /api/articles/:article_id ", () => {
+  test("votes on requested article are updates and returned", async () => {
+    const results = await request(app)
+      .patch("/api/articles/2")
+      .send({ inc_votes: 1 });
+    expect(results.body.article.votes).toEqual(1);
+  });
+  test("votes on requested article are updates and returned", async () => {
+    const results = await request(app)
+      .patch("/api/articles/5")
+      .send({ inc_votes: 1100 });
+    expect(results.body.article.votes).toEqual(1100);
+  });
+  test("votes on requested article are updates and returned - works for negative numbers", async () => {
+    const results = await request(app)
+      .patch("/api/articles/6")
+      .send({ inc_votes: -10 });
+    expect(results.body.article.votes).toEqual(-10);
+  });
   test("returns 404 error if endpoint incorrectly inputted", async () => {
-    const results = await request(app).get("/api/article/2").expect(404);
+    const results = await request(app)
+      .patch("/api/article/6")
+      .send({ inc_votes: -10 })
+      .expect(404);
+    expect(results.body).toEqual({
+      msg: "Page not Found",
+    });
+  });
+  test("returns 404 error if article ID not valid", async () => {
```

Comment:
In this case, the id is valid but not found.

Reply:
Please also add a test for an article with an invalid id (eg '/api/articles/banana')